doc/source/groups_of_atoms.rst

```diff
@@ -64,6 +64,66 @@ Similarly, an :class:`~MDAnalysis.core.groups.Atom` has direct knowledge of the
 
 For information on adding custom Residues or Segments, have a look at :ref:`adding-residue`.
 
+Access to various data views via :class:`~MDAnalysis.core.groups.AtomGroup` object can be pretty powerful, but also
+needs to be used with caution to avoid accessing data outside the intended selection, as mentioned above. Therefore,
+we present two use cases showing commonly used applications.
+
+---------------------------------------
+Use case: Sequence of residues by chain
+---------------------------------------
+Let's assume that we have a PDB file, which contains both ATOM and HETATM record types, for protein and small molecule,
+respectively. In order to select only ATOM record types and get a list of residues by chain, one needs to call:
+
+.. code-block:: python
+
+    u = mda.Universe(pdb, format="PDB")
+    for seg in u.segments:
+        p_seg = seg.atoms.select_atoms("record_type ATOM")
+        r_list = p_seg.residues.resnames
+        print(r_list)
+
+This can be used later on to check whether the residue names belong, for example, to the list of standard residues.
+However, it would be **incorrect** to use:
+
+.. code-block:: python
+
+    selected_atoms = u.select_atoms("record_type ATOM")  # CORRECT SYNTAX, BUT INCORRECT RETURN VALUE!
```

Comment:
Can you please explain why this is an incorrect return value? Phrasing it this way makes it sound like a bug, but it's actually a reasonable step towards `u.select_atoms("record_type ATOM").split("segment")`, which is another way to do this.